cumulusci/utils/xml/metadata_tree.py

```diff
@@ -0,0 +1,265 @@
+    'StandardValueSet'
+
+    Or you can refer to them like this:
+
+    >>> Package["types"]["name"].text
+    'StandardValueSet'
+
+    That might be convenient if you had the key names in a variable or if you had to refer to an element name
+    which would clash with a Python keyword (like `import`) or an instance method (like `text`).
+
+    You can refer to members of lists by index:
+
+    >>> Package["types"]["members"][1]
+    <members>CaseStatus</members> element
+
+    There are also methods for finding, appending, inserting and removing nodes, which have their own documentation.
+    '''
+
+    __slots__ = ["_element", "_parent", "_ns", "tag"]
+
+    def __init__(self, element: etree._Element, parent: etree._Element = None):
+        assert isinstance(element, etree._Element)
+        assert len(element.nsmap) == 1, "Only one namespace allowed"
+        self._element = element
+        self._parent = parent
+        self._ns = next(iter(element.nsmap.values()))
+        self.tag = element.tag.split("}")[1]
+
+    @property
+    def text(self):
+        if len(self._element):  # if self has any element children
+            return self._get_child("text")
+        return self._element.text
+
+    @text.setter
+    def text(self, text):
+        self._element.text = text
+
+    def _wrap_element(self, child: etree._Element):
+        return MetadataElement(child, self._element)
+
+    def _add_namespace(self, tag):
+        return "{%s}%s" % (self._ns, tag)
+
+    def _get_child(self, childname):
+        child_element = self._element.find(self._add_namespace(childname))
+        if child_element is None:
+            raise AttributeError(f"{childname} not found in {self.tag}")
+        return self._wrap_element(child_element)
+
+    def _create_child(self, tag, text=None):
+        element = etree.Element(self._add_namespace(tag))
+        element.text = text
+        return self._wrap_element(element)
+
+    def __getattr__(self, childname):
+        return self._get_child(childname)
+
+    def __getitem__(self, item: Union[int, str]):
+        """You can get either a child-element by name or a sibling element by index.
+
+        The "sibling element" part may seem non-intuitive but it works like this:
+
+        >>> Recipe = fromstring("recipe.xml")
+        >>> Recipe.ingredients[2]
+
+        First it will evaluate `Recipe.ingredients` and generate an Element for the
+        first ingredient.
+
+        Then Python will evaluate the `[2]` and the system will look for the third *sibling*
+        of that first ingredient.
+        """
+        if isinstance(item, int):
+            siblings = self._parent.findall(self._element.tag)
+            return MetadataElement(siblings[item], self._parent)
+        elif isinstance(item, str):
+            return self._get_child(item)
+        else:
+            raise TypeError(
+                f"Indices must be integers or strings, not {type(item)}"
+            )  # # pragma: no cover
```

Comment:
Ugh, so we have code that can be reached at runtime but is untestable due to typeguard :(